listenbrainz/db/model/recommendation_feedback.py

```diff
@@ -0,0 +1,55 @@
+import uuid
+
+from datetime import datetime
+from pydantic import BaseModel, ValidationError, validator
+
+
+def get_allowed_ratings():
+    """ Get rating values that can be submitted corresponding to a recommendation.
+    """
+    return ['like', 'love', 'dislike', 'hate', 'bad_recommendation']
+
+
+def check_recording_mbid_is_valid_uuid(rec_mbid):
+    try:
+        rec_mbid = uuid.UUID(rec_mbid)
+        return str(rec_mbid)
+    except (AttributeError, ValueError):
+        raise ValueError('Recording MBID must be a valid UUID.')
+
+
+class RecommendationFeedbackSubmit(BaseModel):
+    """ Represents a recommendation feedback submit object.
+        Args:
+            user_id: the row id of the user in the DB
+            recording_mbid: the MusicBrainz ID of the recording
+            rating: the feedback associated with the recommendation
+            created: (Optional)the timestamp when the feedback record was inserted into DB
+    """
+
+    user_id: int
+    recording_mbid: str
+    rating: str
+    created: datetime = None
+
+    @validator('rating')
+    def check_feedback_is_valid(cls, rating):
+        expected_rating = get_allowed_ratings()
+        if rating not in expected_rating:
+            raise ValueError('Feedback can only have a value in {}'.format(expected_rating))
+        return rating
+
+    _is_recording_mbid_valid: classmethod = validator("recording_mbid", allow_reuse=True)(check_recording_mbid_is_valid_uuid)
```

Comment:
I think we can use validator as it is like [this](https://github.com/metabrainz/listenbrainz-server/blob/06231b355ad3ce18885b82873c5dd784e7aa70e8/listenbrainz/db/model/feedback.py#L29). If it doesnt work let me know too :p.

Reply:
I wanted to use the same validator function for two classes, this the way for it ig

Reply:
And if you want, you can use the same validation func in feedback. py. It is exactly same func in two places.

Reply:
Lets ask @alastair I think he will have a good point for the clash.